parametric/test_headers_tracestate_dd.py

```diff
@@ -0,0 +1,476 @@
+            test_library,
+            [
+                ["traceparent", "00-12345678901234567890123456789012-1234567890123456-01"],
+                ["tracestate", "foo=1,dd=s:-1;o:tracing2.0"],
+            ],
+        )
+
+    # 1) x-datadog-origin is a well-known value
+    # Result: Origin set to header value
+    assert headers1["x-datadog-origin"] == "synthetics-browser"
+
+    traceparent1, tracestate1 = get_tracecontext(headers1)
+    dd_items1 = tracestate1["dd"].split(";")
+    assert "traceparent" in headers1
+    assert "tracestate" in headers1
+    assert "o:synthetics-browser" in dd_items1
+
+    # 2) x-datadog-origin is NOT a well-known value
+    # Result: Origin set to header value
+    assert headers2["x-datadog-origin"] == "tracing2.0"
+
+    traceparent2, tracestate2 = get_tracecontext(headers2)
+    dd_items2 = tracestate2["dd"].split(";")
+    assert "traceparent" in headers2
+    assert "tracestate" in headers2
+    assert "o:tracing2.0" in dd_items2
+
+    # 3) x-datadog-origin has invalid characters
+    # Result: Origin set to header value, where invalid characters replaced by '_'
+    assert headers3["x-datadog-origin"] == "synthetics;,=web"
+
+    traceparent3, tracestate3 = get_tracecontext(headers3)
+    dd_items3 = tracestate3["dd"].split(";")
+    assert "traceparent" in headers3
+    assert "tracestate" in headers3
+    assert "o:synthetics__~web" in dd_items3
+
+    # 4) tracestate[dd][o] is not present
+    # Result: Origin is not set
+    assert "x-datadog-origin" not in headers4
+
+    traceparent4, tracestate4 = get_tracecontext(headers4)
+    dd_items4 = tracestate4["dd"].split(";")
+    assert "traceparent" in headers4
+    assert "tracestate" in headers4
+    assert not any(item.startswith("o:") for item in dd_items4)
+
+    # 5) tracestate[dd][o] is present and is a well-known value
+    # Result: Origin set to header value
+    assert headers5["x-datadog-origin"] == "synthetics-browser"
+
+    traceparent5, tracestate5 = get_tracecontext(headers5)
+    dd_items5 = tracestate5["dd"].split(";")
+    assert "traceparent" in headers5
+    assert "tracestate" in headers5
+    assert "o:synthetics-browser" in dd_items5
+
+    # 6) tracestate[dd][o] is present and is NOT a well-known value
+    # Result: Origin set to header value
+    assert headers6["x-datadog-origin"] == "tracing2.0"
+
+    traceparent6, tracestate6 = get_tracecontext(headers6)
+    dd_items6 = tracestate6["dd"].split(";")
+    assert "traceparent" in headers6
+    assert "tracestate" in headers6
+    assert "o:tracing2.0" in dd_items6
+
+
+@temporary_enable_propagationstyle_default()
+@pytest.mark.skip_library("dotnet", "Issue: headers5 is not capturing t.dm")
+@pytest.mark.skip_library("golang", "not implemented")
+@pytest.mark.skip_library("nodejs", "TODO: remove when https://github.com/DataDog/dd-trace-js/pull/2477 lands")
+@pytest.mark.skip_library("python", "not implemented")
```

Comment:
I believe the issue with python was the `=` character was not encoded correctly (parsing header2 raised a syntax error).  I believe this bug was fixed. 

Is this test passing now?

Reply:
I just re-enabled the test case and it's passing now 👍🏼 Resolving